### DISCUSSION TITLE
Add Contrast Adaptive Sharpen (CAS) Video Filter

### DISCUSSION THREAD

**Discussion by GT500org** [cdgriffith/FastFlix#448] [Ideas]
I rather like FFmpeg's CAS (more so than other sharpen filters it has). Unfortunately the only way I know of to use it is with the following parameter:
`-vf cas=strength=0.5`

This is incompatible with the -filter_complex parameter that FastFlix uses for its filters, which means that if I want to crop a video or apply any other effects I either can't use CAS at all, or I have to encode a copy with my other filters and then drag that video into FastFlix to encode another copy with CAS applied (which means more passes through the encoder, wasted time, and more quality loss).

It would be great if FastFlix had a built-in option for CAS to help alleviate this problem.

Reasons I like CAS:

1. The visual artifacts/fuzz/etc. produced by CAS are not as obvious as the same amount of sharpening with other filters, and the quality of the sharpening is very good.
2. The performance is _not_ abysmally slow like some other higher quality sharpening filters.

**Comment by cdgriffith** [comment]
Adding to that, it would be good if FastFlix had a way to add custom filters to the `filter_complex` so that any unsupported filter could still be used

**Reply by GT500org** [reply]
That would be cool too.
